frontend/packages/core/src/Input/text-field.tsx

```diff
@@ -162,6 +208,55 @@ export const TextField = ({
     );
   }
 
+  if (isAutoCompleteable) {
+    const [autoCompleteOptions, setAutoCompleteOptions] = React.useState([]);
+    const autoCompleteDebounce = React.useRef(
+      _.debounce(value => {
+        autocompleteCallback(value)
+          .then(data => {
+            setAutoCompleteOptions(data.results);
+          })
+          .catch(err => {
+            helpText = err;
+          });
+      }, 500)
+    ).current;
+
+    // TODO (mcutalo): support option.label in the renderOption
+    return (
+      <Autocomplete
```

Comment:
nit: I would extract this out to a separate functional component

Reply:
Talked in person and were going to pass on this for now.